Apps/WebClient/src/ClientApp/src/models/communication.ts

```diff
@@ -8,9 +8,46 @@ export default interface Communication {
     // Gets or sets the communication text.
     text: string;
 
+    // Gets or sets the type of the Communication.
+    communicationTypeCode: CommunicationType;
+
+    // Gets or sets the state of the Communication.
+    communicationStatusCode: CommunicationStatus;
+
     // Gets or sets the effective date time
     effectiveDateTime: Date;
 
     // Gets or sets the expiry date time
     expiryDateTime: Date;
 }
+
+export enum CommunicationStatus {
+    // New communication.
+    New,
+
+    // Errored out.
+    Error,
+
+    // Communication that has been sent.
+    Processed,
+
+    // Pending communication.
+    Pending,
+
+    // Processing communication.
+    Processing,
+
+    // Draft communication.
+    Draft,
+}
+
+export enum CommunicationType {
+    // Banner communication.
+    Banner,
+
+    // Email communication.
```

Comment:
The front-end at least currently should never see an email communication type.  The code will throw an exception if this occurs on the server so I'm not sure if we should include the Email type.  I was thinking server side that a front end enum could be created but we didn't have a constants package and it didn't seem worth the hassle.